prov/efa/src/rxr/rxr.h

```diff
@@ -441,6 +453,9 @@ struct rxr_tx_entry {
 	/* App-provided reg descriptor */
 	void *desc[RXR_IOV_LIMIT];
 
+	/* Only used for p2p communication with other devices */
+	struct rxr_mr_peer *peer;
```

Comment:
While is this used? (tx_entry->peer)

Reply:
Did you mean where? The data path changes that will come along will use this to tell if we should avoid inline transfers with FI_HMEM_CUDA buffers, and if inline registration should be used or not.

Reply:
Did not we use rxr_is_cuda_mr() to tell if a buffer is cuda or not?

Reply:
```
return rxr_mr ? (rxr_mr->peer.iface == FI_HMEM_CUDA): false
```

You are right, forgot I made that change. We used to key off tx and rx entries, but now can directly use the descriptor. Removing in the next revision.